### PR TITLE
Refactor dashboard, library and media page components

### PR DESCRIPTION
**Pull request by eraychumak**
- Breaks up the cards list component for more flexibility as it was shared across the dashboard, library and search result pages and they all have different purposes and it was getting unnecessarily bloated with route checks and conditional component omission etc
- Viewing library unmatched media is now toggled through the library drop-down menu
- Transitions between library pages are now smoother
- Transitions between media pages are now smoother and layout has changed a tad bit

**Things to do**
- [x] Unmatched media list should hide when switching libraries.
- [x] Fix matching doesn't give any visual feedback
- [x] Switching seasons in media page doesn't work
- [x] Fix up WS events

### CODE REVIEW

File: ui/src/Components/Card/HoverCard.jsx [hunk 1]
```diff
@@ -10,9 +10,9 @@ import CircleIcon from "../../assets/Icons/Circle";
 import SelectMediaFile from "../../Modals/SelectMediaFile/Index.jsx";
 import SelectMediaFilePlayButton from "../../Modals/SelectMediaFile/Activators/PlayButton.jsx";
 
-import "./CardPopup.scss";
+import "./HoverCard.scss";
 
-function CardPopup(props) {
+function HoverCard(props) {
```
Comment (vgarleanu) [inline ui/src/Components/Card/HoverCard.jsx:15]: should probably use a arrow function here for style consistency.

File: ui/src/Components/Card/Index.jsx [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 import { Link } from "react-router-dom";
 
-import CardPopup from "./CardPopup.jsx";
+import HoverCard from "./HoverCard.jsx";
 import Image from "./Image";
 
-import "./Card.scss";
+import "./Index.scss";
 
 function Card(props) {
```
Comment (vgarleanu) [inline ui/src/Components/Card/Index.jsx:9]: Same as above.

File: ui/src/Components/ImageLoad.jsx [hunk 3]
```diff
@@ -30,7 +30,7 @@ function ImageLoad(props) {
     setTryAgain(false);
     setTimeoutID();
 
-    const src = new RegExp("/^(?:/|[a-z]+://)/").test(props.src) ? props.src : `/${props.src}`;
+    const src = new RegExp("^(?:[a-z]+:)?//").test(props.src) ? props.src : `/${props.src}`;
```
Comment (vgarleanu) [inline ui/src/Components/ImageLoad.jsx:33]: whats this modification for?

File: ui/src/Pages/Library/Cards.jsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    try {
+      const config = {
+        headers: {
+          "authorization": auth.token
+        }
+      };
+
+      const res = await fetch(`/api/v1/library/${currentID}/media`, config);
+
+      if (res.status !== 200) {
```
Comment (vgarleanu) [inline ui/src/Pages/Library/Cards.jsx:50]: shouldnt we save the error if its not 200?

**Review by vgarleanu — CHANGES_REQUESTED** [review]
There are some further issues with the library pages.

1. media cards for a library will refresh on WS events even when the events werent meant for that page.
2. library page seems to get remounted on each ws event, causing the posters to redraw and re-animate.
3. when WS events are received, sometimes the library page briefly shows "no media have been found for this library"
4. Options menu animates weirdly when opened, as if it scales last minute in width by a couple of pixels.
5. In the matching menu, files should be sorted by filename.

**Comment by vgarleanu** [comment]
I also think we should allow the user to select media type, but default the media-type to the media-type of the library. Maybe we can hide this under some advanced tab? Sometimes i have anime movies that id like to be in my anime library even tho the anime library type is `tv`.

File: ui/src/Components/ImageLoad.jsx (hunk 3 above)
Comment (eraychumak) [reply ui/src/Components/ImageLoad.jsx:33]: The previous one wasn't detecting an absolute URL properly (e.g. tmdb search result images)

File: ui/src/Pages/Library/Cards.jsx (hunk 4 above)
Comment (eraychumak) [reply ui/src/Pages/Library/Cards.jsx:50]: Errors across the entire app aren't implemented properly right now, haven't come around to 'em yet. For now I simply just check if it was successful or not.

**Comment by eraychumak** [comment]
@vgarleanu need you to pass lib ID of card in the WS events, with the latest commit, the lib page won't flicker anymore but it still fetches media in the background hence I need the ID etc

**Pull request merged**